plugins/auth-backend/api-report.md

```diff
@@ -436,6 +449,20 @@ export type OAuth2ProviderOptions = {
   };
 };
 
+// @public
+export type Oauth2ProxyProviderOptions<JWTPayload> = {
+  authHandler: AuthHandler<OAuth2ProxyResult<JWTPayload>>;
```

Comment:
Is the default behaviour stable enough that we can provide a sensible default here? Like the IAP provider, https://github.com/backstage/backstage/blob/72d4b7061d5f2a1d1d09d77f7aeaf8bd1726e26f/plugins/auth-backend/src/providers/gcp-iap/helpers.ts#L82

Would be nice to be able to make this optional

Reply:
As mentioned [here](https://github.com/backstage/backstage/pull/8396#discussion_r783765260), we tried to add a generic fallback, but there is no real common match. We could go for the `preferred_username` prop, but even that wouldn't work for all cases (see the [docs here](https://github.com/backstage/backstage/pull/8396#discussion_r783765260)).

Would you prefer to have a `defaultAuthHandler` which covers ~70% of the cases instead of having no default?

Reply:
Alright let's keep it required for now 👍  If we wanna provide a default in the future that's a simple non-breaking change

Reply:
Let's make it required and not make a default one. We can always tell people in docs how to add those few lines.

Reply:
👍